File: Leetcode/100.cpp
#include<iostream>
#include<vector>
#include<string>
#include<map>
using namespace std;

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        return sameTree(p,q);
    }

    bool sameTree(TreeNode* p, TreeNode* q){
        if((p==nullptr && q!=nullptr)|| (p!=nullptr && q==nullptr)){
            return false;
        }
        else if(p==nullptr && q==nullptr){
            return true;
        }
        else{
            return (p->val==q->val) && sameTree(p->left,q->left) && sameTree(p->right,q->right);
        }
    }
};

File: Leetcode/100.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    @staticmethod
    def identical(p,q):
        if p==None and q==None:
            return True
        elif (p==None and q!=None) or (p!=None and q==None):
            return False
        elif p.val == q.val:
            return True and Solution.identical(p.left,q.left) and Solution.identical(p.right,q.right)
        elif p.val!=q.val:
            return False
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        return Solution.identical(p,q)
    
